aws_protocol_adaptor/device_client/aws_nvmsgbroker.c

```diff
@@ -243,6 +243,44 @@ NvDsMsgApiErrorType nvds_msgapi_send_async(NvDsMsgApiHandle h_ptr, char *topic,
 	return NVDS_MSGAPI_OK;
 }
 
+/* ************************************************************************* */
+// Subscribe def
+/* ************************************************************************* */
+
+void subscribe_callback_handler(AWS_IoT_Client *pClient, char *topicName, uint16_t topicNameLen,
+								IoT_Publish_Message_Params *params, void *pData)
+{
+	IOT_INFO("Subscribe callback");
+	IOT_INFO("%.*s\t%.*s", topicNameLen, topicName, (int)params->payloadLen, (char *)params->payload);
+}
+
+NvDsMsgApiErrorType nvds_msgapi_subscribe(NvDsMsgApiHandle h_ptr, char **topics, int num_topics, nvds_msgapi_subscribe_request_cb_t cb, void *user_ctx)
+{
+	printf("Subscribe called\n");
+
+	if ((h_ptr == NULL) || (topics == NULL) || (num_topics <= 0))
+	{
+		IOT_ERROR("Essensial args missing for function nvds_msgapi_subscribe: %d, %d, %d\n", (h_ptr == NULL), (topics == NULL), (num_topics == NULL);
+		return NVDS_MSGAPI_ERR;
+	}
+	if (!cb)
+	{
+		IOT_ERROR("Callback function for nvds_msgapi_send cannot be NULL\n");
+		return NVDS_MSGAPI_ERR;
+	}
+	IoT_Error_t rc = FAILURE;
+	AWS_IoT_Client *client = (AWS_IoT_Client *)h_ptr;
+	rc = aws_iot_mqtt_subscribe(client, topics, strlen(topics), QOS0, subscribe_callback_handler, NULL);
```

Comment:
Please review [the documentation](http://aws-iot-device-sdk-embedded-c-docs.s3-website-us-east-1.amazonaws.com/aws__iot__mqtt__client__subscribe_8c.html) for this function.

There are a few problems here:
1. This likely will not compile as you are not using the correct interface for `aws_iot_mqtt_subscribe`
2. `aws_iot_mqtt_subscribe` only subscribes to a single topic (see 1).
3. You're calling `strlen` on an array of strings, not a string.
4. You're not using `cb` at all. Instead you've provided your own callback, so the argument (and by extension, the subscription) is essentially useless.
5. You're not using `user_ctx`